src/config.py

```diff
@@ -18,75 +18,90 @@
-    optimizer_geqo_seed: float | None
-    optimizer_geqo_selection_bias: float | None
-    optimizer_geqo_threshold: int | None
-    optimizer_jit: bool | None
-    optimizer_jit_above_cost: float | None
-    optimizer_jit_inline_above_cost: float | None
-    optimizer_jit_optimize_above_cost: float | None
-    optimizer_join_collapse_limit: int | None
-    optimizer_min_parallel_index_scan_size: int | None
-    optimizer_min_parallel_table_scan_size: int | None
-    optimizer_parallel_setup_cost: float | None
-    optimizer_parallel_tuple_cost: float | None
-    profile: str
-    profile_limit_memory: int | None
+    synchronous_node_count: Literal["all", "majority"] | PositiveInt = Field(default="all")
+    connection_authentication_timeout: int | None = Field(ge=1, le=600, default=None)
+    connection_statement_timeout: int | None = Field(ge=0, le=2147483647, default=None)
+    cpu_parallel_leader_participation: bool | None = Field(default=None)
+    durability_synchronous_commit: Literal["on", "remote_apply", "remote_write"] | None = Field(
+        default=None
+    )
+    durability_wal_keep_size: int | None = Field(ge=0, le=2147483647, default=None)
+    experimental_max_connections: int | None = Field(default=None)
+    instance_default_text_search_config: str | None = Field(default=None)
+    instance_max_locks_per_transaction: int | None = Field(ge=64, le=2147483647, default=None)
+    instance_password_encryption: Literal["md5", "scram-sha-256"] | None = Field(default=None)
+    instance_synchronize_seqscans: bool | None = Field(default=None)
+    ldap_map: str | None = Field(default=None)
+    ldap_search_filter: str | None = Field(default=None)
+    logging_client_min_messages: (
+        Literal[
+            "debug5",
+            "debug4",
+            "debug3",
+            "debug2",
+            "debug1",
+            "log",
+            "notice",
+            "warning",
+            "error",
+        ]
+        | None
+    ) = Field(default=None)
+    logging_log_connections: bool | None = Field(default=None)
+    logging_log_disconnections: bool | None = Field(default=None)
+    logging_log_lock_waits: bool | None = Field(default=None)
+    logging_log_min_duration_statement: int | None = Field(ge=-1, le=2147483647, default=None)
+    logging_track_functions: Literal["none", "pl", "all"] | None = Field(default=None)
+    memory_maintenance_work_mem: int | None = Field(ge=1024, le=2147483647, default=None)
+    memory_max_prepared_transactions: int | None = Field(ge=0, le=262143, default=None)
+    memory_shared_buffers: int | None = Field(ge=16, le=1073741823, default=None)
+    memory_temp_buffers: int | None = Field(ge=100, le=1073741823, default=None)
+    memory_work_mem: int | None = Field(ge=64, le=2147483647, default=None)
+    optimizer_constraint_exclusion: Literal["on", "off", "partition"] | None = Field(default=None)
+    optimizer_cpu_index_tuple_cost: float | None = Field(default=None)
```

Comment:
Missing contraints: `value >= 0 and value <= 1.80e308`

Reply:
Added the validation in https://github.com/canonical/postgresql-operator/pull/902/commits/76d45b923fd7642ed686563a9a211050c27aaca5.